pkg/imgpkg/cmd/copy_repo_src.go

```diff
@@ -60,49 +63,73 @@ func (c CopyRepoSrc) CopyToTar(dstPath string, resume bool) error {
 func (c CopyRepoSrc) CopyToRepo(repo string) (*ctlimgset.ProcessedImages, error) {
 	c.logger.Tracef("CopyToRepo(%s)\n", repo)
 
+	var tempDir string
 	var processedImages *ctlimgset.ProcessedImages
 	importRepo, err := regname.NewRepository(repo)
 	if err != nil {
 		return nil, fmt.Errorf("Building import repository ref: %s", err)
 	}
 
-	if c.TarFlags.IsSrc() {
+	if c.TarFlags.IsSrc() || c.OciFlags.IsOci() {
 		if c.TarFlags.IsDst() {
 			return nil, fmt.Errorf("Cannot use tar source (--tar) with tar destination (--to-tar)")
 		}
 
-		processedImages, err = c.tarImageSet.Import(c.TarFlags.TarSrc, importRepo, c.registry)
+		if c.OciFlags.IsOci() {
+			tempDir, err := os.MkdirTemp("", "imgpkg-oci-extract-")
+			if err != nil {
+				return nil, err
+			}
+			err = image.ExtractOciTarGz(c.OciFlags.OcitoReg, tempDir)
```

Comment:
Let us make sure we clean up after ourselves
```suggestion
                         defer os.RemoveAll(tempDir)
			err = image.ExtractOciTarGz(c.OciFlags.OcitoReg, tempDir)
```